### PR TITLE
DAT-15755 - Sonar issues fix

### PR DESCRIPTION
**Pull request by MalloD12**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [x] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/AbstractLiquibaseMojo.java [hunk 1]
```diff
@@ -875,6 +858,28 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         }
     }
 
+    protected Database getDatabase(ClassLoader mavenClassLoader) throws DatabaseException {
```
Comment (filipelautert) [inline liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/AbstractLiquibaseMojo.java:861]: I think this method could be called something like `createDatabase` and some javadoc would be nice. When I saw `getDatabase ` being used in a `try with resources`  in `AbstractLiquibaseUpdateMojo `at first i tough it was a property getter.

File: liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/AbstractLiquibaseUpdateMojo.java [hunk 2]
```diff
@@ -53,10 +55,11 @@ protected void printSettings(String indent) {
     getLog().info(indent + "number of changes to apply: " + changesToApply);
   }
 
-  protected void handleUpdateException(LiquibaseException exception) throws LiquibaseException {
-    try {
+  protected void handleUpdateException(LiquibaseException exception) throws LiquibaseException, MojoExecutionException {
+    ClassLoader mavenClassLoader = getMavenArtifactClassLoader();
```
Comment (filipelautert) [inline liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/AbstractLiquibaseUpdateMojo.java:59]: Could this ClassLoader getter be moved to `getDatabase/createDatabase` ?

File: liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/LiquibaseHistoryMojo.java [hunk 3]
```diff
@@ -14,10 +16,16 @@ public class LiquibaseHistoryMojo extends AbstractLiquibaseMojo {
 
     @Override
     protected void performLiquibaseTask(Liquibase liquibase) throws LiquibaseException {
-      CommandScope historyCommand = new CommandScope(InternalHistoryCommandStep.COMMAND_NAME);
-
-      historyCommand.addArgumentValue(InternalHistoryCommandStep.DATABASE_ARG, getLiquibase().getDatabase());
-
-      historyCommand.execute();
+        ClassLoader mavenClassLoader;
```
Comment (filipelautert) [inline liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/LiquibaseHistoryMojo.java:19]: Same as above - could this ClassLoader getter be moved to `getDatabase/createDatabase` ? And here we have a try/catch, above we don't.

**Review by filipelautert — COMMENTED** [review]
Overral the removal os duplicated code + usage of try with resources seems good, just some small comments above.

File: liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/AbstractLiquibaseMojo.java (hunk 1 above)
Comment (MalloD12) [reply liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/AbstractLiquibaseMojo.java:861]: Yeah, sure. I created this PR yesterday especially to see if there was something broken I could figure out through the functional tests. What you are proposing here makes total sense to me and I'll apply it now. 

Thanks,
Daniel.

File: liquibase-standard/src/main/java/liquibase/parser/AbstractFormattedChangeLogParser.java [hunk 4]
```diff
@@ -314,7 +314,8 @@
                     } else {
                         try {
                             long ignoreCount = Long.parseLong(ignoreLinesMatcher.group(1));
-                            while (ignoreCount > 0 && reader.readLine() != null) {
+                            String newLine;
```
[the remark this answers is not in the document]
Comment (filipelautert) [reply liquibase-standard/src/main/java/liquibase/parser/AbstractFormattedChangeLogParser.java:317]: Do we need this new line if we re not using it?

File: liquibase-standard/src/main/java/liquibase/snapshot/jvm/PrimaryKeySnapshotGenerator.java [hunk 5]
```diff
@@ -58,7 +58,7 @@ protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot
 
                 String ascOrDesc = row.getString("ASC_OR_DESC");
                 Boolean descending = "D".equals(ascOrDesc) ? Boolean.TRUE : "A".equals(ascOrDesc) ? Boolean.FALSE : null;
-                boolean computed = descending != null && descending;
+                boolean computed = descending != null;
```
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/snapshot/jvm/PrimaryKeySnapshotGenerator.java:61]: This test seems wrong. Descending is being used to hold 3 values here: true, false or null. 
d = true; a = false; else null . Maybe should be changed to an` if ` for clarity

**Review by filipelautert — CHANGES_REQUESTED** [review]

File: liquibase-standard/src/main/java/liquibase/snapshot/jvm/PrimaryKeySnapshotGenerator.java (hunk 5 above)
Comment (MalloD12) [reply liquibase-standard/src/main/java/liquibase/snapshot/jvm/PrimaryKeySnapshotGenerator.java]: Done.

**Review by filipelautert — APPROVED** [review]

**Review by suryaaki2 — APPROVED** [review]

**Pull request merged**